src/arith/num_big.hpp

```
@@ -36,30 +36,35 @@ class Number final {
+        using underlying = std::underlying_type_t<T>;
+        // Note: This does not check that the enum is a valid enum value
+        if (!fits<underlying>()) {
+            CRAB_ERROR("Number ", _n, " does not fit into ", typeid(T).name());
+        }
+        return static_cast<T>(static_cast<underlying>(_n));
     }
 
     template <is_enum T>
-    T cast_to() const {
-        return static_cast<T>(static_cast<std::underlying_type_t<T>>(_n));
+    constexpr T cast_to() const {
```

Clang (14.0.6)

[warning] function 'cast_to' should be marked [[nodiscard]]

(modernize-use-nodiscard)